### PR TITLE
fix decoding of provider error response

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message: fix decoding of provider error response
Additional Description:

There is no response type in the non `Ok` dubbo response. So we should not try to decode response type from non `Ok` dubbo response to avoid the `Cannot parse RpcResult type from buffer` exception. 

Risk Level: Low.
Testing: Added.
Docs Changes: N/A.
Release Notes: N/A.
Platform Specific Features: N/A.

### CODE REVIEW

**Comment by wbpcode** [comment]
cc @zyfjeff

File: test/extensions/filters/network/dubbo_proxy/dubbo_protocol_impl_test.cc [hunk 1]
```diff
@@ -63,6 +65,68 @@ TEST(DubboProtocolImplTest, Normal) {
...
+
+  // Normal dubbo response with exception.
+  {
+    Buffer::OwnedImpl buffer;
+    Buffer::OwnedImpl body_buffer;
+    buffer.add(std::string({'\xda', '\xbb', 0x42, 20}));
+    addInt64(buffer, 1);
+
+    Hessian2::Encoder encoder(std::make_unique<BufferWriter>(body_buffer));
+    encoder.encode<int32_t>(0);
```
Comment (rojkov) [inline test/extensions/filters/network/dubbo_proxy/dubbo_protocol_impl_test.cc:77]: Nit: I presume this line sets an exception in the response body. Could you add a comment about it?

**Review by rojkov — COMMENTED** [review]
Thanks! Looks very good. I've added just one nit.

**Review by rojkov — APPROVED** [review]
Thank you! Looks good to me

**Comment by rojkov** [comment]
/assign-from @envoyproxy/envoy-maintainers 
for a final pass

**Comment by rojkov** [comment]
Oops, unassigning Sotiris...
/assign-from @envoyproxy/envoy-maintainers

**Comment by rojkov** [comment]
The bot is too persistent to assign Sotiris.
Lizan, do you mind giving it a final pass?
/assign @lizan

**Comment by alyssawilk** [comment]
@lizan ping!

**Review by lizan — APPROVED** [review]

**Pull request merged**